CMakeLists.txt

```diff
@@ -96,6 +96,17 @@ if (WITH_MPFR)
     set(HAVE_SYMENGINE_MPFR yes)
 endif()
 
+# Piranha
+set(WITH_PIRANHA no
+    CACHE BOOL "Build with Piranha")
+
+if (WITH_PIRANHA)
+    find_package(Piranha REQUIRED)
+    include_directories(${PIRANHA_INCLUDE_DIRS})
+    set(LIBS ${LIBS} pthread)
```

Comment:
`pthread` is a library that contains threads. It is simply a dependency of PIranha. Another dependency is boost.

So we should simply add the following:
```
    find_package(BOOST REQUIRED)
    include_directories(${BOOST_INCLUDE_DIRS})
    set(LIBS ${LIBS} ${BOOST_LIBRARIES})
```
And the same for `PTHREAD`.

Then we need to add the two packages (`PTHREAD` and `BOOST`) as well. 

The `PTHREAD` package internally can use this: http://stackoverflow.com/questions/15193785/how-to-get-cmake-to-recognize-pthread-on-ubuntu, the `BOOST` package should use our usual library finding code.